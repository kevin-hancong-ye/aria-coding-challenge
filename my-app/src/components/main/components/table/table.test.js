import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react";

import Table from "./table";

test("renders table component", () => {
	render(<Table />);

	const headingElem = screen.getByRole("heading");
	expect(headingElem).toHaveTextContent("Table");

	const elem = screen.getByRole("region");
	expect(elem).toBeInTheDocument();
	expect(elem).toMatchSnapshot();
});
