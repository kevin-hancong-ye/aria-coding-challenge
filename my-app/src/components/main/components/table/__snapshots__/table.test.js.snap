// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders table component 1`] = `
<div
  aria-labelledby="table-heading"
  id="table_section"
  role="region"
>
  <h2
    id="table-heading"
    role="heading"
  >
    Table
  </h2>
  <table
    aria-label="Expenses"
    class="test"
    role="table"
  >
    <tbody>
      <tr
        class="grayBackground"
      >
        <th>
          Company
        </th>
        <th>
          Contact
        </th>
        <th>
          Country
        </th>
      </tr>
      <tr
        aria-rowindex="0"
      >
        <td>
          companyA
        </td>
        <td>
          contactA
        </td>
        <td>
          countryA
        </td>
      </tr>
      <tr
        aria-rowindex="1"
      >
        <td>
          companyB
        </td>
        <td>
          contactB
        </td>
        <td>
          countryB
        </td>
      </tr>
      <tr
        aria-rowindex="2"
      >
        <td>
          companyC
        </td>
        <td>
          contactC
        </td>
        <td>
          countryC
        </td>
      </tr>
      <tr
        aria-rowindex="3"
      >
        <td>
          companyD
        </td>
        <td>
          contactD
        </td>
        <td>
          countryD
        </td>
      </tr>
      <tr
        aria-rowindex="4"
      >
        <td>
          companyE
        </td>
        <td>
          contactE
        </td>
        <td>
          countryE
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
